components/extensions/org.wso2.carbon.ml.siddhi.extension/src/main/java/org/wso2/carbon/ml/siddhi/extension/PredictStreamProcessor.java

```diff
@@ -203,6 +274,25 @@ private void populateFeatureAttributeMapping() throws Exception {
         }
     }
 
+    private Object getMostFrequent(Object[] predictions) {
```

Comment:
Get rid of this method and use commons-math library or equivalent, which gives you the mode.
https://commons.apache.org/proper/commons-math/apidocs/org/apache/commons/math3/stat/StatUtils.html#mode%28double[]%29